### PR TITLE
xtensa/esp32: Add ESP32 EFUSE char driver

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Add ESP32 EFUSE char driver
## Impact
ESP32 users will be able to read/burn the EFUSEs
## Testing
Tested on esp32-wrover-kit board

NOTE: Coding Style check will fail because mixed case issues

### CODE REVIEW

File: include/nuttx/efuse/esp_efuse.h [hunk 1]
```diff
@@ -0,0 +1,184 @@
+/****************************************************************************
+ * include/nuttx/efuse/esp_efuse.h
```
Comment (xiaoxiang781216) [inline include/nuttx/efuse/esp_efuse.h:2]: Can we define a generic efuse driver interface? So not only ESP but also other chip can implement this interface.

File: arch/xtensa/src/esp32/esp32_efuse.c [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+{
+  sem_t   exclsem;            /* Supports mutual exclusion */
+  bool    batch_writing_mode; /* Controls batch writing */
+};
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct file_operations g_efusefops =
```
Comment (xiaoxiang781216) [inline arch/xtensa/src/esp32/esp32_efuse.c:60]: Like the previous comment, it's better to extract g_efusefops to an upper half driver.

Comment (acassis) [reply arch/xtensa/src/esp32/esp32_efuse.c:60]: Hi @xiaoxiang781216 I agree with these modifications to support a generic efuse driver. Just moved this PR to Draft

**Comment by sdc-g** [comment]
Hi @acassis 
Thanks for your commit and I see this PR is still in draft status.
It is appreciate if the esp32 board test app or example could be shared after modification, thank you very much!
BRs
Feng

**Comment by acassis** [comment]
> Hi @acassis
> Thanks for your commit and I see this PR is still in draft status.
> It is appreciate if the esp32 board test app or example could be shared after modification, thank you very much!
> BRs
> Feng

Thank you @sdc-g I will submit the example so you can use it with the current draft example

**Comment by sdc-g** [comment]
> driver

No hurry, @acassis , it is OK if the example based on the final version.
Thanks for your quick response!

**Comment by acassis** [comment]
> > driver
> 
> No hurry, @acassis , it is OK if the example based on the final version.
> Thanks for your quick response!

I already did it: https://github.com/apache/incubator-nuttx-apps/pull/526

**Comment by acassis** [comment]
I'll submit the new generic driver

**Pull request closed without merging**